internal/xdscache/v3/endpointslicetranslator.go

```diff
@@ -61,7 +61,15 @@ func (c *EndpointSliceCache) RecalculateEndpoints(port, healthPort v1.ServicePor
 					continue
 				}
 
-				if p.Protocol == nil || (healthPort.Protocol != *p.Protocol || port.Protocol != *p.Protocol) && *p.Protocol != v1.ProtocolTCP {
+				if p.Protocol == nil {
+					continue
+				}
+
+				if *p.Protocol != v1.ProtocolTCP {
+					continue
+				}
+
+				if port.Protocol != *p.Protocol || healthPort.Protocol != *p.Protocol {
 					continue
 				}
```

Comment:
Same comment here, digging into the unit test failures, I think it's fine to drop this if statement here. This code is intended to just skip over any endpoints ports that are not using TCP protocol, since that's all we support. During the DAG build we already ensure that healthPort.Protocol and port.Protocol are TCP, so as long as we're filtering out endpoint ports with non-TCP protocol (which we're doing above), then this check on L61-63 is not needed.